Adding chapter to Content Management Guide on Convert2RHEL process (#674)
theforeman/foreman-documentation: guides/common/assembly_converting-a-host-with-convert2rhel.adoc

ifdef::context[:parent-context: {context}]

[id="assembly_converting-a_host-with-convert2rhel_{context}"]
= Converting a Host to {RHEL} with Convert2RHEL

Convert2RHEL enables the conversion of Red Hat Enterprise Linux derivative distributions into a supportable {RHEL} state while retaining existing applications and configurations.

The conversion process is similar to a minor release upgrade of {RHEL} in which every RPM package on the system is replaced.
Third-party packages and non-{Team} packages not available in {RHEL} are not replaced.

Convert2RHEL removes unnecessary packages such as logos, or packages known to cause issues during the conversion, and replaces the `CentOS-release` package with the `rhel-release` package.

> It subscribes the system to {Team} Subscription Management and {ProjectName} and replaces all packages signed by CentOS.

Reviewer: ```suggestion
It subscribes the system to {Team} Subscription Management and {ProjectName} and replaces all packages signed by CentOS.
```

`{Team} Subscription Management ` only works for Satellite docs, IMHO. Can you reword it maybe?

Author: Fixed.

The amount of time taken to do this can vary depending on how many packages need to be replaced, network speed, storage speed, and similar variables.

Use the global registration template to register and subscribe your system before the conversion.

> For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/6.9/html/managing_hosts/registering_hosts#registering-a-host-to-project-using-the-global-registration-template_managing-hosts[Registering a Host to {ProjectName} Using the Global Registration Template] in the _Managing Hosts_ guide.

Reviewer: ```suggestion
For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/{AccessRedHatComVersion}/html/managing_hosts/registering_hosts#registering-a-host-to-project-using-the-global-registration-template_managing-hosts[Registering a Host to {ProjectName} Using the Global Registration Template] in the _Managing Hosts_ guide.
```

To convert hosts to {RHEL} with convert2RHEL, you must complete the following procedures:

. xref:installing-convert2rhel_{context}[].
. xref:creating-a-host-activation-key_{context}[].
. xref:preparing-the-host-group_{context}[].
. xref:registering-hosts-for-conversion_{context}[].

> . xref:converting-hosts-with-convert2rhel_{context}[

Author: Fixed

.Prerequisites
* You must have completed the steps listed in https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/converting_from_an_rpm-based_linux_distribution_to_rhel/index#proc_preparing-for-a-rhel-conversion_converting-from-a-linux-distribution-to-rhel[Preparing for a RHEL conversion].

> * If you are providing the organization ID and activation key to register and subscribe your system, you must have created an activation key in {ProjectName} or {Team} Subscription Manager.

Reviewer: See comment above regarding `{Team} Subscription Manager`.

Author: Fixed.

> For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/6.9/html/content_management_guide/managing_activation_keys[Managing Activation Keys] in this guide, and https://access.redhat.com/documentation/en-us/red_hat_subscription_management/1/html-single/using_red_hat_subscription_management/index#activation_keys_con[Understanding activation keys] in {Team} Subscription Manager documentation.

Reviewer: ```suggestion
For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/{AccessRedHatComVersion}/html/content_management_guide/managing_activation_keys[Managing Activation Keys] in this guide, and https://access.redhat.com/documentation/en-us/red_hat_subscription_management/1/html-single/using_red_hat_subscription_management/index#activation_keys_con[Understanding activation keys] in {Team} Subscription Manager documentation.
```

> * Ensure you have a subscription manifest uploaded to your Satellite and that there are sufficient {RHEL} subscriptions for the conversion.

Reviewer: ```suggestion
* Ensure you have a subscription manifest uploaded to your {Project} and that there are sufficient {RHEL} subscriptions for the conversion.
```

ifdef::context[:parent-context: {context}]

[id="assembly_converting-a_host-with-convert2rhel_{context}"]
= Converting a Host to {RHEL} with Convert2RHEL

Convert2RHEL enables the conversion of Red Hat Enterprise Linux derivative distributions into a supportable {RHEL} state while retaining existing applications and configurations.

The conversion process is similar to a minor release upgrade of {RHEL} in which every RPM package on the system is replaced.
Third-party packages and non-{Team} packages not available in {RHEL} are not replaced.

> Convert2RHEL removes unnecessary packages such as logos, or packages known to cause issues during the conversion, and replaces the `CentOS-release` package with the `rhel-release` package.

Reviewer: Question: Does this only work from "centos" to "rhel"? What about Oracle Linux, Alma Linux, and Rocky Linux, which are binary/bug compatible with RHEL?

Author: I'm told they are not included  in this release. They'll be included at or after 7.0

It subscribes the system to {Team} Subscription Management and {ProjectName} and replaces all packages signed by CentOS.

The amount of time taken to do this can vary depending on how many packages need to be replaced, network speed, storage speed, and similar variables.

Use the global registration template to register and subscribe your system before the conversion.
For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/6.9/html/managing_hosts/registering_hosts#registering-a-host-to-project-using-the-global-registration-template_managing-hosts[Registering a Host to {ProjectName} Using the Global Registration Template] in the _Managing Hosts_ guide.

To convert hosts to {RHEL} with convert2RHEL, you must complete the following procedures:

. xref:installing-convert2rhel_{context}[].
. xref:creating-a-host-activation-key_{context}[].
. xref:preparing-the-host-group_{context}[].
. xref:registering-hosts-for-conversion_{context}[].
. xref:converting-hosts-with-convert2rhel_{context}[].

> == Prerequisites
> 

Reviewer: ```suggestion
.Prerequisites
```

Author: Accepted and changed.

* You must have completed the steps listed in https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/converting_from_an_rpm-based_linux_distribution_to_rhel/index#proc_preparing-for-a-rhel-conversion_converting-from-a-linux-distribution-to-rhel[Preparing for a RHEL conversion].
* If you are providing the organization ID and activation key to register and subscribe your system, you must have created an activation key in {ProjectName} or {Team} Subscription Manager.
For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/6.9/html/content_management_guide/managing_activation_keys[Managing Activation Keys] in this guide, and https://access.redhat.com/documentation/en-us/red_hat_subscription_management/1/html-single/using_red_hat_subscription_management/index#activation_keys_con[Understanding activation keys] in {Team} Subscription Manager documentation.
* Ensure you have a subscription manifest uploaded to your Satellite and that there are sufficient {RHEL} subscriptions for the conversion.

[NOTE]
====

> If you are accessing {RHEL} packages using {ProjectServer}, you must use the organization ID and activation key to run the convert2rhel command.

Reviewer: ```suggestion
If you are accessing {RHEL} packages using {ProjectServer}, you must use the organization ID and activation key to run the `convert2rhel` command.
```

Author: Accepted and changed.

* Ensure you have a subscription manifest uploaded to your Satellite and that there are sufficient {RHEL} subscriptions for the conversion.
* If you are providing the organization ID and activation key to register and subscribe your system, you must have created an activation key in {ProjectName} or {Team} Subscription Manager.
For more information, see https://access.redhat.com/documentation/en-us/red_hat_satellite/{AccessRedHatComVersion}/html/content_management_guide/managing_activation_keys[Managing Activation Keys] in this guide, and https://access.redhat.com/documentation/en-us/red_hat_subscription_management/1/html-single/using_red_hat_subscription_management/index#activation_keys_con[Understanding activation keys] in {Team} Subscription Manager documentation.
* Ensure you have a subscription manifest uploaded to your {Project} and that there are sufficient {RHEL} subscriptions for the conversion.

[NOTE]
====
If you are accessing {RHEL} packages using {ProjectServer}, you must use the organization ID and activation key to run the `convert2rhel` command.

> ====

Reviewer 2: If you can avoid NOTE, avoid note markup. (Brno as well)

Author: Fixed
